src/CFNetwork/MessageHandler.cs

```diff
@@ -103,15 +105,15 @@ CFHTTPMessage CreateRequest (HttpRequestMessage request, bool isFirstRequest)
 
 			SetupRequest (request, message, isFirstRequest);
 
-			if ((auth == null) || (Credentials == null) || !PreAuthenticate)
+			if ((auth is null) || (Credentials is null) || !PreAuthenticate)
 				return message;
 
-			if (!auth.AppliesToRequest (message))
+			if (auth is not null && !auth.AppliesToRequest (message))
 				return message;
 
-			var method = auth.GetMethod ();
+			var method = auth?.GetMethod ();
 			var credential = Credentials.GetCredential (request.RequestUri, method);
```

Comment:
Does the Credentials.GetCredential take a null value in the method? That is something to look at before we land this since we now know that method can be null.

Reply:
That is a good point.
Because these files are not in the xamios.csproj (or frameworks.sources), I added them to the xamios.csproj just to handle the nullability since visual studio underlines them, but there are a lot of things that Visual Studio can't tell me and one of those things is what the heck the Credentials variable is. Thus I cannot find the method for Credentials.GetCredential... I still don't know where these files are being used, but building main does complain if I don't fix all the nullability
![image](https://user-images.githubusercontent.com/50846373/157335954-b64d9a52-f33d-476a-b69d-5c4f1a38bedc.png)

Reply:
There was a reason why I asked ;) 

This class is extending the HttpClientHandler, that is a System.Net class, so you will not find the source in our tree, you can use the reference code here: https://github.com/microsoft/referencesource/blob/master/System/net/System/Net/Http/HttpClientHandler.cs

Credentials is a ICrendential implementation, which is using https://docs.microsoft.com/en-us/dotnet/api/system.net.icredentials.getcredential?view=net-6.0 The issue here is that, the signature of the interface does not let us know if it is nullable, in that case, I wold not consider it safe to send a null parameter to a contract (interface) that does not explicitly say that it takes null. Also, the "method" parameter is https://docs.microsoft.com/en-us/dotnet/api/system.net.iauthenticationmodule.authenticationtype?view=net-6.0#system-net-iauthenticationmodule-authenticationtype which is the http auth type values, which AFAIK does not take null (I can only remember a few: Basic, Digest, Ntml, Kerberos and Anonymous, maybe null == anonymous? ).

I would not do this changes since it looks like you added a number of possible NRE based in the auth? being possibly null, in this particular case, we need to check if null and do the correct thing rather than use ? in the method calls.